tests/kafkatest/services/verifiable_consumer.py

```diff
@@ -0,0 +1,215 @@
+        self.kafka = kafka
+        self.topic = topic
+        self.group_id = group_id
+        self.max_messages = max_messages
+        self.session_timeout = session_timeout
+
+        self.assignment = {}
+        self.joined = set()
+        self.total_records = 0
+        self.consumed_positions = {}
+        self.committed_offsets = {}
+        self.revoked_count = 0
+        self.assigned_count = 0
+
+        for node in self.nodes:
+            node.version = version
+
+        self.prop_file = ""
+        self.security_config = kafka.security_config.client_config(self.prop_file)
+        self.prop_file += str(self.security_config)
+
+    def _worker(self, idx, node):
+        node.account.ssh("mkdir -p %s" % VerifiableConsumer.PERSISTENT_ROOT, allow_fail=False)
+
+        # Create and upload log properties
+        log_config = self.render('tools_log4j.properties', log_file=VerifiableConsumer.LOG_FILE)
+        node.account.create_file(VerifiableConsumer.LOG4J_CONFIG, log_config)
+
+        # Create and upload config file
+        self.logger.info("verifiable_consumer.properties:")
+        self.logger.info(self.prop_file)
+        node.account.create_file(VerifiableConsumer.CONFIG_FILE, self.prop_file)
+        self.security_config.setup_node(node)
+
+        cmd = self.start_cmd(node)
+        self.logger.debug("VerifiableConsumer %d command: %s" % (idx, cmd))
+
+        for line in node.account.ssh_capture(cmd):
+            event = self.try_parse_json(line.strip())
+            if event is not None:
+                with self.lock:
+                    name = event["name"]
+                    if name == "offsets_committed":
+                        self._handle_offsets_committed(node, event)
+                    elif name == "records_consumed":
+                        self._handle_records_consumed(node, event)
+                    elif name == "partitions_revoked":
+                        self._handle_partitions_revoked(node, event)
+                    elif name == "partitions_assigned":
+                        self._handle_partitions_assigned(node, event)
+
+    def _handle_offsets_committed(self, node, event):
+        if event["success"]:
+            for offset_commit in event["offsets"]:
+                topic = offset_commit["topic"]
+                partition = offset_commit["partition"]
+                tp = TopicPartition(topic, partition)
+                self.committed_offsets[tp] = offset_commit["offset"]
+
+    def _handle_records_consumed(self, node, event):
+        for topic_partition in event["partitions"]:
+            topic = topic_partition["topic"]
+            partition = topic_partition["partition"]
+            tp = TopicPartition(topic, partition)
+            self.consumed_positions[tp] = topic_partition["maxOffset"] + 1
+        self.total_records += event["count"]
+
+    def _handle_partitions_revoked(self, node, event):
+        self.revoked_count += 1
+        self.assignment[node] = []
```

Comment:
Maybe we can also check if the revoked partitions are equal to the current assigned partitions.